### PR TITLE
User authentication setup

### PR DESCRIPTION
**Pull request by Emelit0**
added minimal requirements for userModel,
routes for register and login,
controller to handle register and login,
mongodb connection to test both register and login.

password is stored as hash in database and user gets send a jsonwebtoken.

**Not sure if sending back the hashed password to user is a security risk?**

### CODE REVIEW

File: server/index.js
Comment (lugenx) [inline server/index.js:38]: a database connection is already maintained in `95-db-and-models-setup` branch and is currently in a PR.

File: server/package.json
Comment (lugenx) [inline server/package.json:20]: since it's not decided to add nodemon as a dependency, i recommend you install it globally if you want to use it

**Review by lugenx — CHANGES_REQUESTED** [review]
thank you for your effort and interest in the project. 

i added some comments, but a lot of code in this pr is already provided by another pr, and it's already in progress, this will have to wait until PR #99 is merged. 

for future reference, to avoid this kind of issue before starting the work it's a good idea to get assigned to the issue first and make sure nobody is working on that part of the project.

thanks again 👍

File: server/src/controller/authController.js
Comment (AbhiBajpaishek) [inline server/src/controller/authController.js:31]: can we use status code 401 for Invalid Credentials?

File: server/src/controller/authController.js
Comment (lugenx) [inline server/src/controller/authController.js:20]: yes, i believe there is a bit of a security risk there. it's better to verify users at the backend first, then sign a token, and send email and token to the client instead of email and hashed password.

File: server/src/controller/authController.js
Comment (Emelit0) [reply server/src/controller/authController.js:31]: 401 Unauthorized is the status code to return when the client provides no credentials or invalid credentials

**Comment by AbhiBajpaishek** [comment]
@Emelit0  Could you please look into merge conflicts? Also, if its easier, pls create a new branch and PR and scrap this one. Please refer below issue which was created for this work. I've added few comments in there

https://github.com/lugenx/ecohabit/issues/96

**Pull request closed without merging**
